plugins/content-manager/src/main/java/com/wazuh/contentmanager/index/ContextIndex.java

```diff
@@ -148,15 +178,29 @@ public void onFailure(Exception e) {
     public ConsumerInfo getConsumer(String context, String consumer) {
         try {
             SearchResponse searchResponse = get(context).get(TIMEOUT, TimeUnit.SECONDS);
+            log.info("Received search response for {}: {}", context, searchResponse.status());
+
             Map<String, Object> source =
                     (Map<String, Object>)
                             searchResponse.getHits().getHits()[0].getSourceAsMap().get(consumer);
-            Long last_offset = (Long) source.get(ConsumerInfo.LAST_OFFSET);
+            if (source == null) {
+                log.warn("Consumer {} not found in context {}", consumer, context);
+                return null;
+            }
+
+            Object offsetObj = source.get(ConsumerInfo.LAST_OFFSET);
+            Long last_offset = (offsetObj instanceof Number) ? ((Number) offsetObj).longValue() : null;
             String snapshot = (String) source.get(ConsumerInfo.LAST_SNAPSHOT_LINK);
             return new ConsumerInfo(consumer, context, last_offset, snapshot);
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+        } catch (TimeoutException e) {
+            log.error("Timeout retrieving context [{}], consumer [{}]", context, consumer, e);
+        } catch (Exception e) {
             log.error(
-                    "Failed to retrieve context [{}], consumer [{}]: {}", context, consumer, e.getMessage());
+                    "Failed to retrieve context [{}], consumer [{}]: {}",
+                    context,
+                    consumer,
+                    e.getMessage(),
+                    e);
         }
         return null;
```

Comment:
Here again, we could raise errors instead of returning null.

Reply:
Applied on https://github.com/wazuh/wazuh-indexer-plugins/pull/307/commits/9d2c816113f51654e9d6f818f56391f89f7608b0